File: ephi_healthcare_worker_app/lib/models/patient.dart
class Patient {
  String id;
  String firstName;
  String lastName;
  String phoneNumber;
  String createdAt;
  String updatedAt;
  String dateOfBirth;
  String status;
  List<dynamic> history;
  String email;
  String woreda;
  String userId;
  String language;
  String gender;
  Patient(
      {this.id,
      this.firstName,
      this.lastName,
      this.phoneNumber,
      this.createdAt,
      this.updatedAt,
      this.history,
      this.email,
      this.woreda,
      this.userId,
      this.dateOfBirth,
      this.status,
      this.language,
      this.gender});
  factory Patient.fromJson(Map<String, dynamic> json) {
    return Patient(
      id: json['_id'] as String,
      firstName: json['first_name'] as String,
      gender: json['gender'] as String,
      lastName: json['last_name'] as String,
      phoneNumber: json['phone_number'] as String,
      createdAt: json['created_at'] as String,
      updatedAt: json['updated_at'] as String,
      status: json['status'] as String,
      language: json['language'] as String,
      dateOfBirth: json['dob'] as String,
      //history: List.from(json['history']),
      email: json['email'] as String,
      woreda: json['woreda'] as String,
      userId: json['user_id'] as String,
    );
  }
}

File: Backend/models/Patient.js
const mongoose = require("mongoose");

const patientSchema = new mongoose.Schema({
  _id: mongoose.Schema.Types.ObjectId,

    user_id: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true,
        index: true
    },
    first_name: {
        type: String,
        required: true
    },
    last_name: {
        type: String,
        required: true
    },
    dob: {
        type: Date,
    },
    phone_number: {
        type: String,
    },
    email:{
        type: String,
    },
    language: {
        type: String,
    },
    gender: {
        type: String,
        enum: ["MALE", "FEMALE", "UNDISCLOSED"],
        required: true,
    },
    woreda: {
        type: String,
    },
    street_address: {
        type: String,
    },
    city: {
        type: String,
    },
    state: {
        type: String,
    },
    sms_status: {
        type: Boolean,
    },
    status:{
        type: String,
        enum: ["Unknown","Recovered","Confirmed","Died"],
        required: true,
        default: "Unknown"
    },
    history:[],
    emergency_contact:{
        first_name: {
            type: String,
        },
        last_name: {
            type: String,
        },
        relationship:{
            type: String,
        },
        city:{
            type: String,
        },
        state: {
            type: String,
        },
        phone_number:{
            type: String,
        },
    }

},
{
    timestamps: { createdAt: 'created_at', updatedAt: 'updated_at' }
});

const patientSchemaDemo = new mongoose.Schema({
    _id: mongoose.Schema.Types.ObjectId,
  
      user_id: {
          type: mongoose.Schema.Types.ObjectId,
          ref: 'Demo User',
          required: true,
          index: true
      },
      first_name: {
          type: String,
          required: true
      },
      last_name: {
          type: String,
          required: true
      },
      dob: {
          type: Date,
      },
      phone_number: {
          type: String,
      },
      email:{
          type: String,
      },
      language: {
          type: String,
      },
      gender: {
          type: String,
          enum: ["MALE", "FEMALE", "UNDISCLOSED"],
          required: true,
      },
      woreda: {
          type: String,
      },
      street_address: {
          type: String,
      },
      city: {
          type: String,
      },
      state: {
          type: String,
      },
      sms_status: {
          type: Boolean,
      },
      status:{
          type: String,
          enum: ["Unknown","Recovered","Confirmed","Died"],
          required: true,
          default: "Unknown"
      },
      history:[],
      emergency_contact:{
          first_name: {
              type: String,
          },
          last_name: {
              type: String,
          },
          relationship:{
              type: String,
          },
          city:{
              type: String,
          },
          state: {
              type: String,
          },
          phone_number:{
              type: String,
          },
      }
  },
  {
      timestamps: { createdAt: 'created_at', updatedAt: 'updated_at' }
  });
const Patient = mongoose.model("Patient", patientSchema);
const PatientDemo = mongoose.model("PatientDemo", patientSchemaDemo);
exports.Patient = Patient;
exports.PatientDemo = PatientDemo;
